dspace-api/src/main/java/org/dspace/uclouvain/profileIngester/ProfileIngesterCLI.java

```diff
@@ -0,0 +1,195 @@
+ * The event action type present in Rabbit determines the action to perform on the researcher profile.
+ * This CLI can be executed multiple times together to increased processing rate.
+ * 
+ * @author Michaël Pourbaix (michael.pourbaix@uclouvain.be)
+ */
+public class ProfileIngesterCLI extends AbstractCLICommand {
+
+    private static final Option OPT_PERSON = Option.builder("e")
+        .longOpt("eperson")
+        .hasArg(true)
+        .desc("email address of eperson doing importing.")
+        .required(true)
+        .build();
+    private static final Option OPT_ERROR_QUEUE_NAME = Option.builder("eq")
+        .longOpt("error_queue")
+        .hasArg(true)
+        .desc("RabbitMQ error queue name to store potential errors. (Defaults to {queue_name + '_error'})")
+        .required(false)
+        .build();
+    public static final String USAGE_DESCRIPTION =
+        "A command-line tool to listen for person event and process corresponding action in DSpace.";
+
+    private static final Logger logger = LoggerFactory.getLogger(ProfileIngesterCLI.class);
+
+    // EXTERNAL SERVICES
+    protected EPersonService ePersonService;
+    protected ResearcherProfileService researcherProfileService;
+    protected ConfigurationService configService;
+    protected PersonEventConnector personEventConnector;
+
+    // COMMON CLICommand METHODS----------------------------------
+    @Override
+    protected void buildOptions() {
+        serviceOptions.addOption(OPT_PERSON);
+        serviceOptions.addOption(OPT_ERROR_QUEUE_NAME);
+        infoOptions.addOption(OPT_HELP);
+    }
+
+    @Override
+    protected String getUsageDescription() {
+        return USAGE_DESCRIPTION;
+    }
+
+    // METHODS----------------------------------------------------
+    public static void main(String[] arguments) throws Exception {
+        ProfileIngesterCLI profileIngester = new ProfileIngesterCLI();
+        CommandLine cli = profileIngester.validateCLIArgument(arguments);
+
+        PersonEventConnector connector = new PersonEventConnector();
+
+        String epersonEmail = cli.getOptionValue("e");
+        String errorQueueName = cli.getOptionValue("eq", connector.getQueueName() + "_error");
+        while (true) {
+            try {
+                profileIngester.ingestEvent(epersonEmail, connector, errorQueueName);
+            } catch (InterruptedException ie) {
+                logger.info("Execution interrupted manually");
+                Thread.currentThread().interrupt();
+                break;
+            } catch (Exception e) {
+                logger.error(e.getMessage(), e);
+            }
+        }
+    }
+
+    ProfileIngesterCLI() throws IOException, TimeoutException {
+        ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
+        researcherProfileService = new DSpace().getSingletonService(ResearcherProfileService.class);
+    }
```

Comment:
Best practice: Construtor should be defined after attributes and before methods (I think...)